shared-bindings/watchdog/WatchDogMode.c

```diff
@@ -24,76 +24,33 @@
  * THE SOFTWARE.
  */
 
+#include "py/enum.h"
+
 #include "shared-bindings/watchdog/WatchDogMode.h"
 
+MAKE_ENUM_VALUE(watchdog_watchdogmode_type, watchdogmode, NONE, WATCHDOGMODE_NONE);
+MAKE_ENUM_VALUE(watchdog_watchdogmode_type, watchdogmode, RAISE, WATCHDOGMODE_RAISE);
+MAKE_ENUM_VALUE(watchdog_watchdogmode_type, watchdogmode, RESET, WATCHDOGMODE_RESET);
+
 //| class WatchDogMode:
-//|     """run state of the watchdog timer"""
-//|
-//|     def __init__(self) -> None:
-//|         """Enum-like class to define the run mode of the watchdog timer."""
-//|     RAISE: WatchDogMode
-//|     """Raise an exception when the WatchDogTimer expires.
+//|     """Run state of the watchdog timer."""
 //|
-//|     **Limitations:** ``RAISE`` mode is not supported on SAMD or RP2040.
+//|     NONE: WatchDogMode
+//|     """The `WatchDogTimer` is disabled."""
 //|
-//|     :type WatchDogMode:"""
+//|     RAISE: WatchDogMode
+//|     """Raise an exception when the `WatchDogTimer` expires."""
 //|
 //|     RESET: WatchDogMode
-//|     """Reset the system if the WatchDogTimer expires.
-//|
-//|     :type WatchDogMode:"""
+//|     """Reset the system when the `WatchDogTimer` expires."""
 //|
-const mp_obj_type_t watchdog_watchdogmode_type;
-
-const watchdog_watchdogmode_obj_t watchdog_watchdogmode_raise_obj = {
-    { &watchdog_watchdogmode_type },
+MAKE_ENUM_MAP(watchdog_watchdogmode) {
+    MAKE_ENUM_MAP_ENTRY(watchdogmode, NONE),
+    MAKE_ENUM_MAP_ENTRY(watchdogmode, RAISE),
+    MAKE_ENUM_MAP_ENTRY(watchdogmode, RESET),
 };
```

Comment:
I think this isn't really equivalent, because before the `None` object was used, but now it's `WatchDogMode.NONE`. Am I mistaken, or does this prevent existing code from working if it tries to check for the None watchdogmode?

Reply:
The `None` object wasn't used before. `WatchDogMode.NONE` is a replacement for `deinit`.

Reply:
```py
Adafruit CircuitPython 8.2.6 on 2023-09-12; Adafruit PyGamer with samd51j19
>>> import microcontroller
>>> microcontroller.watchdog.mode is None
True
```
In 8.2.x, `None` is the value of the `mode` property before it is set to something else. After this PR, it is `WatchDogMode.NONE`.

I see this as an incompatibility. It may be minor enough that we just go ahead and do it. @tannewt @dhalbert I would appreciate your opinion on how closely we require backward compatibility.

I would be willing to see if the "enum" code (which I wrote) can be enhanced to use "None" as the Python value for the 0-enumerant, which might fix this incompatibility.

Reply:
https://gist.github.com/jepler/596da0c2c488f51d19d547c6cc8116aa

Reply:
I see. The earlier implementation used `WATCHDOGMODE_NONE` enum internally and `None` object for the python side.

https://github.com/adafruit/circuitpython/blob/1c0155c99b757a1c7840dd6cf6486b1aedfb99d1/shared-bindings/watchdog/WatchDogMode.c#L56-L75

Using the `None` object has two benefits:

1. One less QSTR.
2. Maintains backward compatibility.

Reply:
I have updated the PR to use the `None` object. Can you test it.